### PR TITLE
Ups 889 robo wallet can send solana token lockup transactions

### PR DESCRIPTION
**Pull request by makarychev**
Update interfaces regarding changes in TokenLockup ABI 
1. added new api call initializeTimelock
2. changed api call FundReleaseSchedule

### CODE REVIEW

File: lib/TokenLockupHelper.mjs [hunk 1]
```diff
@@ -259,4 +265,40 @@ export default class TokenLockupTxHelper {
...
+      [tokenlockAccount.toBuffer(), target.toBuffer()],
+      this.program.programId,
+    )
+    return timelockAccount
+  }
+
+  async getTimelockAccountData(timelockAccount) {
+    try {
+      return await this.program.account.timelockData.fetch(timelockAccount)
+    } catch (_error) {
```
[the remark this answers is not in the document]
Comment (makarychev) [reply lib/TokenLockupHelper.mjs:280]: I would like to say that It is not an error. If account data doesn't exist we will need to create it

File: lib/splTockenlockupAbi.js [hunk 2]
```diff
@@ -592,352 +489,83 @@ const TOKEN_LOCKUP_ABI = {
...
       'name': 'InvalidTokenlockAccount',
-      'msg': 'invalid tokenlock account data'
+      'msg': 'Invalid tokenlock account data'
     },
     {
       'code': 6001,
-      'name': 'StateSizeLimited',
-      'msg': 'state data size reached to max'
+      'name': 'MaxReleaseDelayLessThanOne',
+      'msg': 'Max release delay must be greater than 1'
```
[the remark this answers is not in the document]
Comment (makarychev) [reply lib/splTockenlockupAbi.js:503]: You are right. Message will be updated

File: lib/TokenLockupHelper.mjs [hunk 3]
```diff
@@ -88,25 +90,26 @@ export default class TokenLockupTxHelper {
...
+    const timelockAccount = await this.getTimelockAccount(tokenAccPubKey, toPubKey)
+    const timelockAccountData = await this.getTimelockAccountData(timelockAccount)
     let instructions = []
-    if (getAssociatedTokenAddressInstr !== null) {
-      instructions.push(getAssociatedTokenAddressInstr)
+    if (timelockAccountData === null) {
+      instructions.push(this.initializeTimelockInstruction(tokenAccPubKey, timelockAccount, toPubKey, fromPubKey))
     }
 
+    const mintToken = await this.createMint(obj.tokenMintAddress, tokenAccPubKey)
```
[the remark this answers is not in the document]
Comment (makarychev) [reply lib/TokenLockupHelper.mjs:100]: It is very rare case and as I found Solana Web3js is usually throw understandable messages

File: lib/TokenLockupHelper.mjs [hunk 4]
```diff
@@ -88,25 +90,26 @@ export default class TokenLockupTxHelper {
     const cancelableBy = obj.cancelableBy.map(element => new PublicKey(element))
     const nonceBytes = getNonceBytes(new BigNumber(obj.nonce))
 
-    const mintToken = await this.createMint(obj.tokenMintAddress, tokenAccPubKey)
-    const fromTokenAccount = await mintToken.getOrCreateAssociatedAccountInfo(fromPubKey)
-    const [toTokenAccount, getAssociatedTokenAddressInstr] =
-      await this.getAssociatedTokenAddress(mintToken, toPubKey, fromPubKey)
-
+    const timelockAccount = await this.getTimelockAccount(tokenAccPubKey, toPubKey)
```
[the remark this answers is not in the document]
Comment (makarychev) [reply lib/TokenLockupHelper.mjs:93]: In most cases we want to stop transfer so it is caught inside https://github.com/CoMakery/robowallet/blob/main/lib/blockchains/SolanaBlockchain.js#L170

**Review by Alexey1100 — APPROVED** [review]

**Pull request merged**